pymeasure/instruments/jobinyvon/jobinyvon270m.py

```diff
@@ -0,0 +1,340 @@
+        """
+        self.write_read(b'\xF8')
+        self.write_read(b'\xDE')
+
+    def auto_baud(self):
+        """
+        This function tries to set the spectrometer into
+        intelligent mode (mode that allows communication with the
+        instrument). The logic of this function is based on the
+        spectrometer's datasheet. If the function successfully sets
+        the instrument into intelligent mode, we return True,
+        False otherwise.
+        """
+        while True:
+            ans = self.write_read(b' ')
+            if len(ans) != 0:
+                break
+
+        code = self._get_code(ans)
+        if code != 'F':
+            if code == "*":
+                ans = self.write_read(b'\xF7')
+                code = self._get_code(ans)
+                if code != '=':
+                    return False
+                ans = self.write_read(b' ')
+                code = self._get_code(ans)
+
+            if code == "B":
+                self.write_read(b'O2000\0')
+                time.sleep(0.5)
+                ans = self.write_read(b' ')
+                code = self._get_code(ans)
+                if code == 'F':
+                    return True
+                elif code == ' ':
+                    self.write_read(b'\xF8')
+                    ans = self.write_read(b' ')
+                    code = self._get_code(ans)
+                    if code == 'F':
+
+                        return True
+                    else:
+                        self.unstuck()
+                        return False
+                else:
+                    self.unstuck()
+                    return False
+            else:
+                self.unstuck()
+                return False
+        return True
+
+    def motor_init(self) -> bool:
+        """
+        Spectrometer initialization function.
+        """
+        ans = self.write_read(b'A', nread=1, timeout=100000)
+        code = self._get_code(ans)
+        if code != 'o':
+            raise IOError(f'Wrong return code from driver, received {code}')
+        return True
+
+    def move_grating_steps(self, nsteps: int):
+        """
+        ABSOLUTE positioning of the grating motor in number of steps.
+        """
+        ans = self.write_read(f'F0,{nsteps - self.grating_steps}\r'.encode(), nread=1, timeout=20000)
+        code = self._get_code(ans)
+        if code != 'o':
+            raise IOError(f'Wrong return code from driver, received {code}')
+
+    def get_grating_wavelength(self):
+        """
+        Reading the wavelength from the grating motor of the spectrometer.
+        """
+        return self._lambda_max - ((self._max_steps - self.grating_steps) / self._steps_nm)
+
+    def move_grating_wavelength(self, wavelength: float):
+        """
+        ABSOLUTE positioning of the grating motor in wavelength.
+        """
+        steps = int(self._max_steps - int((self._lambda_max - wavelength) * self._steps_nm))
+        self.move_grating_steps(steps)
```

Comment:
Those functions you could make a property:
```suggestion
    @property
    def grating_wavelength(self):
        """
        Reading the wavelength from the grating motor of the spectrometer.
        """
        return self._lambda_max - ((self._max_steps - self.grating_steps) / self._steps_nm)

    @grating_wavelengt.setter
    def grating_wavelength(self, wavelength: float):
        """
        ABSOLUTE positioning of the grating motor in wavelength.
        """
        steps = int(self._max_steps - int((self._lambda_max - wavelength) * self._steps_nm))
        self.move_grating_steps(steps)
```

Reply:
Done